dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/WorkFlowLineageMapper.xml

```diff
@@ -161,4 +161,58 @@
         </if>
         AND a.task_type = 'DEPENDENT'
     </select>
+
+    <select id="queryTaskDepOnProcess" resultType="org.apache.dolphinscheduler.dao.entity.TaskMainInfo">
+        select td.id
+            , td.name as taskName
+            , td.code as taskCode
+            , td.version as taskVersion
+            , td.task_type as taskType
+            , ptr.process_definition_code as processDefinitionCode
+            , pd.name as processDefinitionName
+            , pd.version as processDefinitionVersion
+            , pd.release_state as processReleaseState
+        from t_ds_task_definition td
+        join t_ds_process_task_relation ptr on ptr.post_task_code = td.code and td.version = ptr.post_task_version
+        join t_ds_process_definition pd on pd.code = ptr.process_definition_code and pd.version = ptr.process_definition_version
+        <where>
+            <if test="projectCode != 0">
+                and ptr.project_code = #{projectCode}
+            </if>
+            <if test="processDefinitionCode != 0">
+                and ptr.process_definition_code != #{processDefinitionCode}
+                and (
+                    td.task_params like concat('%"definitionCode":', #{processDefinitionCode}, '%')
```

Comment:
:crying_cat_face: why we had two ways....

Reply:
it can be use in both subprocess(`'%"processDefinitionCode":', #{processDefinitionCode}, '%'`) and dependent(`'%"definitionCode":', #{processDefinitionCode}, '%'`)

Reply:
it can be use in both subprocess(`'%"processDefinitionCode":', #{processDefinitionCode}, '%'`) and dependent(`'%"definitionCode":', #{processDefinitionCode}, '%'`)

Reply:
It's too complex in the mapper, can we handle it in the program?

Reply:
@lenboo  I find some exists logic about how to query task definition content in like `td.task_params like concat('%', #{processDefinitionCode}, '%')`. It is almost the same complex, and I think it may query some wrong record in the return result, because we may have tasks that have `processDefinitionCode` in the task param in the further

Reply:
and if we query without `td.task_params like concat('%"definitionCode":', #{processDefinitionCode}, '%') or td.task_params like concat('%"processDefinitionCode":', #{processDefinitionCode}, '%')`, will almost query all the task relation from database, which I think will overload the database

Reply:
This SQL looks complex because it does a lot of things. 
It is recommended to separate SQL queries.

Reply:
rebase to fix test failed